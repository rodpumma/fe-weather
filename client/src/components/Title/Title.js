import React from 'react'

const Title = ({text}) => {
  return (
    <pre data-testid='title' className="weather-card__day-header">
      {text}
    </pre>
  )
}

export default Title
