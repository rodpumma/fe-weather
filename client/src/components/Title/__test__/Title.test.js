import React from 'react'
import ReactDOM from 'react-dom'
import Title from '../Title'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

afterEach(cleanup)
describe('<Title />', () => {
  it('Check <Title /> render', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Title />, div)
  })
  
  it('Check <Title /> render with text prop', () => {
    const { getByTestId } = render(<Title text='City name' />)
    expect(getByTestId('title')).toHaveTextContent('City name')
  })
  
  it('<Title /> matches snapshop', () => {
    const tree = renderer
      .create(<Title text='City Name' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
