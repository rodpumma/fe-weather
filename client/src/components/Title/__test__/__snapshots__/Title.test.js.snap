// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> <Title /> matches snapshop 1`] = `
<pre
  className="weather-card__day-header"
  data-testid="title"
>
  City Name
</pre>
`;
